tests/deploy/test_batch_job.py

```diff
@@ -91,6 +93,209 @@ def test_extract_result_metadata_aggregate_spatial_delayed_vector():
+    # This smaller rectangle is also the BBox that we expect in the metadata.
+    aggrgeo_west_latlon = 4.2
+    aggrgeo_east_latlon = 4.8
+    aggrgeo_south_latlon = 51.2
+    aggrgeo_north_latlon = 51.8
+    geometries_lat_lon = Polygon(
+        (
+            (aggrgeo_west_latlon, aggrgeo_south_latlon),
+            (aggrgeo_west_latlon, aggrgeo_north_latlon),
+            (aggrgeo_east_latlon, aggrgeo_north_latlon),
+            (aggrgeo_east_latlon, aggrgeo_south_latlon),
+            (aggrgeo_west_latlon, aggrgeo_south_latlon),
+        )
+    )
+    cube = cube.aggregate_spatial(geometries=geometries_lat_lon, reducer="mean")
+
+    metadata = extract_result_metadata(tracer)
+    # Allow a 1% margin in the approximate comparison of the BBox
+    expected_bbox = [
+        approx(aggrgeo_west_latlon, 0.01),
+        approx(aggrgeo_south_latlon, 0.01),
+        approx(aggrgeo_east_latlon, 0.01),
+        approx(aggrgeo_north_latlon, 0.01),
+    ]
+    assert metadata["bbox"] == expected_bbox
+    assert metadata["area"] == {
+        "value": approx(6763173869883.0, 1.0),
+        "unit": "square meter",
+    }
+
+
+@pytest.mark.parametrize(
+    ["crs_epsg", "west", "south", "east", "north"],
+    [
+        # BBoxes below correspond to lat-long from 4E 51N to 5E 52N.
+        # Belgian Lambert 2008
+        (3812, 624112.728540544, 687814.368911342, 693347.444114525, 799212.044310798),
+        # ETRS89 / UTM zone 31N (N-E)
+        (3043, 570168.861511006, 5650300.78652147, 637294.365895774, 5762926.81279022),
+        # Netherlands, Amersfoort / RD New
+        (28992, 57624.6287650174, 335406.866285557, 128410.08537081, 445806.50883315),
+    ],
+)
+def test_extract_result_metadata_aggregate_spatial_delayed_vector_when_bbox_crs_not_epsg4326(
+    tmp_path, crs_epsg, west, south, east, north
+):
+    tracer = DryRunDataTracer()
+    cube = tracer.load_collection(
+        collection_id="Sentinel2",
+        arguments={
+            "temporal_extent": ["2020-02-02", "2020-03-03"],
+        },
+    )
+    cube = cube.filter_bbox(
+        west=west, south=south, east=east, north=north, crs=crs_epsg
+    )
+    cube.resample_spatial(resolution=0, projection=crs_epsg)
+
+    #
+    # Create a geojson file for the spatial aggregation with a delayed vector file.
+    # GeoJSON is always in lat-long, so we don't reproject this geometry.
+    #
+    # To keep the bbox numbers sensible we create a rectangular polygon that is
+    # slightly smaller than the BBox in filter_bbox above, so that they will stay
+    # within the valid bounds of the (projected) CRS of the cube.
+    # That way, we should not get large and/or inaccurate coordinates if any
+    # CRS conversion has happened along the way.
+    #
+    # This smaller rectangle is also the BBox that we expect in the metadata.
+    aggrgeo_west_latlon = 4.2
+    aggrgeo_east_latlon = 4.8
+    aggrgeo_south_latlon = 51.2
+    aggrgeo_north_latlon = 51.8
+    geometries_lat_lon = Polygon(
```

Comment:
FYI: shapely has a helper for creating a rectangle like this: `shapely.geometry.box`  https://shapely.readthedocs.io/en/1.8.5/manual.html?highlight=box#shapely.geometry.box

Reply:
Great tip, thanks!
Done.